js-runtime/index.js

```diff
@@ -1051,6 +1051,53 @@ module.exports = {
       return [ false, 'namespace-key pair not found', ]
     }
   },
+  dsrrun: async (nskey, func) => {
+    const val = ds[`${nskey[0]}:${nskey[1]}`];
+    return [true, await func(val)];
+  },
+  dsmrun: async (nskey, func) => {
+    let val = ds[`${nskey[0]}:${nskey[1]}`];
+    const [out, newval] = await func(val);
+    ds[`${nskey[0]}:${nskey[1]}`] = newval;
+    return [true, out];
+  },
+  dsrwith: async (wth, func) => {
+    const nskey = wth[0];
+    const a = wth[1];
+    const b = ds[`${nskey[0]}:${nskey[1]}`];
+    return [true, await func(a, b)];
+  },
+  dsmwith: async (wth, func) => {
+    const nskey = wth[0];
+    const a = wth[1];
+    let b = ds[`${nskey[0]}:${nskey[1]}`];
+    // Get out, newb!
+    const [out, newb] = await func(a, b);
+    ds[`${nskey[0]}:${nskey[1]}`] = newb;
+    return [true, out];
+  },
+  dsmonly: async (nskey, func) => {
+    let val = ds[`${nskey[0]}:${nskey[1]}`];
+    const newval = await func(val);
+    ds[`${nskey[0]}:${nskey[1]}`] = newval;
+  },
+  dswonly: async (wth, func) => {
```

Comment:
I'd rather call it `dsmwonly` to follow more or less the same pattern in all opcodes

Reply:
I can't remember if the opcode name limit is 7 or 8 characters, so I went with 7.

Reply:
I remember it actually being 7, but I can't find anywhere in the avm/compiler where it would get shortened to 7 (and they would need to agree in order for the opcode hashmap to work). It must've been fixed at some point